lib/index.ts

```diff
@@ -0,0 +1,40 @@
+import * as bodyParser from 'body-parser';
+import * as express from 'express';
+import * as _ from 'lodash';
+
+import {NetworkManager} from './nm';
+import wifiApi from './routes/wifi';
+
+interface Route {
+	method: string;
+	handler: (nm: NetworkManager) => (req: any, res: any) => void;
+}
+
+function start(port: number = 1337) {
+	const app = express();
+	app.use(bodyParser.json());
+	const nm = new NetworkManager();
+
+	return nm.init()
+	.then(() => {
+		const exposeNmApiToApp = exposeAPIs(app)(nm);
+		exposeNmApiToApp('/wifi')(wifiApi);
+		app.listen(port);
+	})
+	.catch((err) => {
+		throw err;
+	});
+}
+
+function exposeAPIs(app) {
+	return (nm: any) => (rootApi: string = '/') => (APIs: any) => {
+		const root: string = (rootApi === '/') ? '' : rootApi;
+		_.each(APIs, (route: Route, path: string) => {
+			return app[route.method.toLowerCase()](`${root}/${path}`, route.handler(nm));
+		});
+	};
+}
+
+export {
+	start,
```

Comment:
I think it'd be good to expose the nm api itself, rather than just the server, and export the server separately (i.e. `var nm = require('nm-api')` & then have `nm.HttpServer` or so), in order to make this usable as a package that can be consumed by other modules without having to use the HTTP interface. Just a nit though, as it's not required for Etcher Pro.

Reply:
Yep, I forgot to add that and it will let people use the module even without the HTTP interface, which is extremely useful.
Adding it now